versioned/tiered/mongodb/src/main/java/org/projectnessie/versioned/mongodb/MongoFragment.java

```diff
@@ -30,21 +33,24 @@
 
   static void produce(Document document, Fragment v) {
     produceBase(document, v)
-        .keys(deserializeKeys(document, KEY_LIST));
+        .keys(deserializeKeysWithEntityType(document, KEY_LIST));
```

Comment:
@stevelorddremio @kylep-dremio mongo also changed a bit. I added the entity type field here